src/Sylius/Bundle/CoreBundle/Form/Type/Product/ChannelPricingType.php

```diff
@@ -35,7 +36,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $form = $event->getForm();
 
                 $form->add('price', MoneyType::class, [
-                    'label' => $channel->getName(),
+                    'label' => 'Price for ' . $channel->getName(),
```

Comment:
As for translation issues:

We can set label for `price` field to "sylius.ui.price", label for `originalPrice` field to "sylius.ui.original_price" and then pass `$channel` to form view to allow for custom rendering. Then we can overwrite Twig block for channel pricing (after #7664 is merged) to something like:

```html
<fieldset>
  <legend>{{ channel.name }}</legend>
  {{ form_row(channelPricing.price) }}
  {{ form_row(channelPricing.originalPrice) }}
</fieldset>
```

If you can't find time to dig into it further, tell me so I'll make a PR to your branch.